src/hints/mod.rs

```diff
@@ -336,3 +340,24 @@ pub fn breakpoint(
     println!("\n-----------END BREAKPOINT-----------\n");
     Ok(())
 }
+
+#[allow(unused)]
+const SET_AP_TO_ACTUAL_FEE: &str = "memory[ap] = to_felt_or_relocatable(execution_helper.tx_execution_info.actual_fee)";
+pub fn set_ap_to_actual_fee(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    _ids_data: &HashMap<String, HintReference>,
+    _ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let execution_helper = exec_scopes.get::<ExecutionHelperWrapper>("execution_helper")?;
+    let actual_fee = execution_helper
+        .execution_helper
+        .borrow()
+        .tx_execution_info
```

Comment:
Could you add a `fn tx_execution_info(&self) -> Result<&TransactionExecutionInfo, ExecutionHelperError>` method to `ExecutionHelper`?

Reply:
Hmmm, that doesn't really make sense to me because it's an `Option<TransactionExecutionInfo>` which expresses that the TEI may or may not be present (depending on context, specifically whether we are between `start_tx()` and `end_tx()`, I believe).

Also, there currently isn't any `impl ExecutionHelper`, all the "helpers" exist in `impl ExecutionHelperWrapper`. Something like this might fit in there, but there are a lot of fields you could potentially do this for, which would be a lot of code bloat. I still don't think a `Result` makes sense, either.

Reply:
This is the model used for the `DeprecatedOsSyscallHandlerWrapper`. I agree that it's a lot of bloat but all that RC + RefCell unwrapping should remain an implementation detail IMO.